src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -4273,3 +4273,29 @@ void C2_MacroAssembler::vector_mask_operation(int opc, Register dst, XMMRegister
   vector_mask_operation_helper(opc, dst, tmp, masklen);
 }
 #endif
+
+void C2_MacroAssembler::vector_maskall_operation(KRegister dst, Register src, int mask_len) {
+  if (VM_Version::supports_avx512bw()) {
+    if (mask_len > 32) {
+      kmovql(dst, src);
+      kshiftrql(dst, dst, 64 - mask_len);
```

Comment:
Here masklen is 64, so kshiftrql is not needed here?

Reply:
src operand carry either a 0 (false mask) or 0XFFFFFFFFFFFFFFFF  (true mask) value. Thus a right shift here ensures that destination bits whose size matches with mask_len are set.

I have introduced special case handling when mask_len is  64, 32 and 16.